client/sysinfo/src/sysinfo.rs

```diff
@@ -32,14 +33,117 @@ use std::{
 	time::{Duration, Instant},
 };
 
+/// Used to represent the unit used in the benchmarks.
+pub enum Unit {
+	GiBs,
+	MiBs,
+	KiBs,
+}
+
+impl fmt::Display for Unit {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		f.write_str(match self {
+			Unit::GiBs => "GiBs",
+			Unit::MiBs => "MiBs",
+			Unit::KiBs => "KiBs",
+		})
+	}
+}
+
+/// Throughput as measured in bytes per second.
+#[derive(Debug, Clone, Copy, PartialEq, PartialOrd)]
+pub struct Throughput(f64);
+
+const KIBIBYTE: f64 = 1024.0;
+
+impl Throughput {
+	/// `f64` kibibyte/s to byte/s.
+	pub fn from_kibs(kibs: f64) -> Throughput {
```

Comment:
```suggestion
	/// Construct [`Self`] from kibibyte/s.
	pub fn from_kibs(kibs: f64) -> Throughput {
```
... analogous for the functions below.